src/main.js

```diff
@@ -100,8 +102,10 @@ class ImageKitUppyPlugin extends Plugin {
 
         this.uppy.log('[ImageKit] Uploading...')
         const files = fileIDs.map((fileID) => this.uppy.getFile(fileID))
-
-        return this.uploadFiles(files).then(() => null)
+        const filesFiltered = filterNonFailedFiles(files)
+        const filesToEmit = filterFilesToEmitUploadStarted(filesFiltered)
+        this.uppy.emit('upload-start', filesToEmit)
```

Comment:
This event is already emitted for individual files in `uploadFiles` when upload "starts". And isn't it after the upload ends? Why are we adding it here in `handleUpload`?

Also I didn't get the use of these two [filters](https://github.com/transloadit/uppy/blob/main/packages/%40uppy/utils/src/fileFilters.js) here.

Reply:
removed the filters need to emit this event before uploading starts for all the files otherwise uppy was giving this  [issue ](https://www.loom.com/share/25c234c8277746ecb34d7767ef0f74b6?sid=0cb8c1ec-8c2a-4168-a1c5-2d2c2a117b6d)
we can find the refrence here in xhr-upload code

Reply:
So this event emitted in `_uploadRemote` is not needed?

Reply:
yes i think but i am not able to test for remote because it wont work locally and on stage packages or not live